LLMonFHIR/LLMonFHIRStandard.swift

```diff
@@ -40,100 +34,83 @@ actor LLMonFHIRStandard: Standard, HealthKitConstraint, EnvironmentAccessible {
     
     
+    @MainActor
     func fetchRecordsFromHealthKit() async {
-        guard await useHealthKitResources else {
-            await MainActor.run {
-                waitingState.isWaiting = false
-            }
+        await waitingState.run {
+            await _fetchRecordsFromHealthKit()
+        }
+    }
+    
+    @MainActor
+    private func _fetchRecordsFromHealthKit() async {
+        guard useHealthKitResources else {
             return
         }
-        
         await fhirStore.removeAllResources()
-        
-        await MainActor.run {
-            waitingState.isWaiting = true
-        }
-        await triggerWaitingTask()
-        
-        let healthKit = self.healthKit
+        let healthKit = await healthKit
         await withTaskGroup { taskGroup in
             for recordType in Self.recordTypes {
-                taskGroup.addTask { [self] in
+                taskGroup.addTask { @concurrent [self] in
+                    dispatchPrecondition(condition: .notOnQueue(.main))
                     let records = try? await healthKit.query(
                         recordType,
                         timeRange: .ever,
                         limit: self.resourceLimit,
                         sortedBy: [SortDescriptor(\.startDate, order: .reverse)]
                     )
-                    
                     guard let records else {
                         return
                     }
-                    
                     await addRecords(records)
                 }
             }
         }
+        await updateSchemas()
     }
     
     private func addRecords(_ records: [HKClinicalRecord]) async {
-        await withTaskGroup { sampleTaskGroup in
-            for newHealthKitSample in records {
-                sampleTaskGroup.addTask { [self] in
-                    await triggerWaitingTask()
+        await withTaskGroup { taskGroup in
+            for record in records {
+                taskGroup.addTask { [self] in
                     do {
-                        try await fhirStore.add(newHealthKitSample, loadHealthKitAttachments: true)
+                        try await fhirStore.add(record, loadHealthKitAttachments: true)
                     } catch {
-                        logger.error("Could not transform sample \(newHealthKitSample.id) to FHIR resource: \(error)")
+                        logger.error("Could not transform sample \(record.id) to FHIR resource: \(error)")
                     }
                 }
             }
         }
-    }
-    
-    @FHIRProcessingActor
-    private func triggerWaitingTask() async {
-        waitTask?.cancel()
-        waitTask = Task {
-            try? await Task.sleep(for: .seconds(10))
-            
-            if !Task.isCancelled {
-                await MainActor.run {
-                    waitingState.isWaiting = false
-                }
-                await updateSchemas()
-            }
-        }
+        await updateSchemas()
```

Comment:
Only a small thing; but one of the benefits beforehand was that we only updated the schema at the every end, and not on every addRecords call in the `_fetchRecordsFromHealthKit` loop. Wondering if we might need to slightly differentiate between the call of the function here from within `_fetchRecordsFromHealthKit` and the one if we would get new health records while the application is running and e.g. HealthKit adds more records?